### PR TITLE
update LinkTabs: optional useReactLink prop and required name prop

### PR DESCRIPTION
**Pull request by hwigmore**
[Clubhouse](https://app.clubhouse.io/floored/story/14665/add-optional-usereactlink-prop-to-linktabs)

### CODE REVIEW

File: docs/_javascript/TabPreview.jsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ class TabPreview extends React.Component {
       <div>
         { getPreviewComponent('Horizontal',
             <RadioTabs
+              name="horizontal-tabs"
```
Comment (hwigmore) [inline docs/_javascript/TabPreview.jsx]: this change is unrelated to this pr, but `name` was recently made `required`, so updating the react component previews

File: react/tabs/LinkTabs.jsx [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
+
+  return (
+    <ul className={`${props.isVertical ? 'blx-vertical-tabs' : 'blx-horizontal-tabs'} ${props.isDisabled ? 'blx-disabled' : ''}`}>
+      { links }
+    </ul>
+  );
+};
 
 LinkTabs.propTypes = {
+  name: PropTypes.string.isRequired,
```
Comment (hwigmore) [inline react/tabs/LinkTabs.jsx]: similar to the recent change in `RadioTabs`, i made `name` a required field. this way, we can use a concatenation of name and each link's text to get a `key` for each item in the list. this way, if there are multiple instances of the same link tabs on the same page, we can give them unique names like "top-nav-links", "left-nav-links" etc and have all keys be unique.

File: react/tabs/LinkTabs.jsx [hunk 3]
```diff
@@ -1,40 +1,42 @@
...
-          </a>
-        </li>
-      ))
-    }
-  </ul>
-);
+const LinkTabs = (props) => {
+  const links = props.tabs.map((tab, idx) => {
+    const classes = `blx-tab-item ${idx === props.activeTabIdx ? 'blx-active' : 'blx-inactive'}`;
+    const link = props.useReactLink ?
```
Comment (hwigmore) [inline react/tabs/LinkTabs.jsx]: this ternary is the main change

File: packages/blocks-docs/_javascript/TabPreview.jsx [hunk 4]
```diff
@@ -22,6 +22,7 @@ class TabPreview extends React.Component {
       <div>
         { getPreviewComponent('Horizontal',
             <RadioTabs
+              name="horizontal-tabs"
```
Comment (hwigmore) [inline packages/blocks-docs/_javascript/TabPreview.jsx]: this change is unrelated to this pr, but `name` was recently made `required`, so updating the react component previews

File: packages/blocks-react/tabs/LinkTabs.jsx [hunk 5]
```diff
@@ -1,40 +1,42 @@
...
+
+  return (
+    <ul className={`${props.isVertical ? 'blx-vertical-tabs' : 'blx-horizontal-tabs'} ${props.isDisabled ? 'blx-disabled' : ''}`}>
+      { links }
+    </ul>
+  );
+};
 
 LinkTabs.propTypes = {
+  name: PropTypes.string.isRequired,
```
Comment (hwigmore) [inline packages/blocks-react/tabs/LinkTabs.jsx]: similar to the recent change in `RadioTabs`, i made `name` a required field. this way, we can use a concatenation of name and each link's text to get a `key` for each item in the list. this way, if there are multiple instances of the same link tabs on the same page, we can give them unique names like "top-nav-links", "left-nav-links" etc and have all keys be unique.

File: packages/blocks-docs/_javascript/TabPreview.jsx (hunk 4 above)
Comment (hwigmore) [inline packages/blocks-docs/_javascript/TabPreview.jsx]: this change is unrelated to this pr, but `name` was recently made `required`, so updating the react component previews

File: packages/blocks-react/tabs/LinkTabs.jsx [hunk 6]
```diff
@@ -1,40 +1,42 @@
...
-          </a>
-        </li>
-      ))
-    }
-  </ul>
-);
+const LinkTabs = (props) => {
+  const links = props.tabs.map((tab, idx) => {
+    const classes = `blx-tab-item ${idx === props.activeTabIdx ? 'blx-active' : 'blx-inactive'}`;
+    const link = props.useReactLink ?
```
Comment (hwigmore) [inline packages/blocks-react/tabs/LinkTabs.jsx]: this ternary is the main change

**Comment by hwigmore** [comment]
@MagdalenaHenke @tariqporter ping ping on this one

also adding @emma-floored

**Comment by hwigmore** [comment]
@hamham91 wanna take a look at this? would allow us to remove one more override when we update blocks

**Comment by hamham91** [comment]
looks good! 👍

**Comment by tariqporter** [comment]
👍

**Pull request merged**